packages/blip/app/components/chart/deviceUsage.js

```diff
@@ -0,0 +1,214 @@
+import CardContent from "@material-ui/core/CardContent";
+import CardHeader from "@material-ui/core/CardHeader";
+import Divider from "@material-ui/core/Divider";
+import Grid from "@material-ui/core/Grid";
+import Table from "@material-ui/core/Table";
+import TableBody from "@material-ui/core/TableBody";
+import TableCell from "@material-ui/core/TableCell";
+import TableContainer from "@material-ui/core/TableContainer";
+import TableRow from "@material-ui/core/TableRow";
+import Typography from "@material-ui/core/Typography";
+
+import PhonelinkSetupOutlinedIcon from "@material-ui/icons/PhonelinkSetupOutlined";
+
+import { BasicsChart } from "tideline";
+import Stats from "./stats";
+import { getParametersChanges, getLongDayHourFormat, formatParameterValue } from "tidepool-viz";
+
+const useStyles = makeStyles((theme) => ({
+  card: {
+    width: 430,
+  },
+  cardHeader: {
+    textTransform: "uppercase",
+    backgroundColor: "var(--card-header-background-color)",
+  },
+  cardContent: {
+    overflowY: "auto",
+    maxHeight: 800,
+  },
+  sectionTitles: {
+    fontSize: "var(--section-title-font-size)",
+    fontWeight: "var(--section-title-font-weight)",
+    lineHeight: "var(--section-title-line-height)",
+    margin: "var(--section-title-margin)",
+    color: "var(--section-title-color)"
+  },
+  sectionContent: {
+    fontSize: "13px",
+    fontWeight: 300,
+    lineHeight: "15px",
+    color: "#444444"
+  },
+  deviceLabels: {
+    paddingLeft: theme.spacing(2),
+  },
+  tableRows: {
+    "&:nth-of-type(odd)": {
+      backgroundColor: theme.palette.grey[100],
+    },
+  },
+  tableCell: {
+    padding: theme.spacing(0.2, 0.5),
+  },
+  divider: {
+    margin: theme.spacing(1, 0),
+    width: "99%",
+  },
+  parameterChanges: {
+    width: "100%",
+  },
+  parameterChangesTable: {
+    maxHeight: 200,
+  },
+}));
+
+const getLabel = (row, t) => {
+  const fCurrentValue = `${formatParameterValue(row.value, row.unit)} ${row.unit}`;
+  const currentLabel = t(`params|${row.name}`);
+  switch (row.changeType) {
+  case "added":
+    return `${currentLabel} (${fCurrentValue})`;
+  case "deleted":
+    return `${currentLabel} (${fCurrentValue} -> ${t("deleted")})`;
+  case "updated":
+    const fPreviousValue = `${formatParameterValue(row.previousValue, row.previousUnit)} ${row.unit}`;
+    return `${currentLabel} (${fPreviousValue} -> ${fCurrentValue})`;
+  default:
+    return `${currentLabel} X (${fCurrentValue})`;
+  }
+};
+
+const DeviceUsage = (props) => {
+  //eslint-disable-next-line
+  const { bgPrefs, timePrefs, patient, tidelineData, permsOfLoggedInUser, trackMetric,
+    //eslint-disable-next-line
+    dataUtil, chartPrefs, endpoints, loading
+  } = props;
+  const { t } = useTranslation();
+  const classes = useStyles();
+  //eslint-disable-next-line
+  const mostRecentSettings = _.last(tidelineData.grouped.pumpSettings);
+  const device = _.get(mostRecentSettings, "payload.device", null);
```

Comment:
opt: Wouldn't something like this be the same: `const device = mostRecentSettings?.payload?.device;`?
This would avoid using loadash